linchpin/api/fetch/fetch.py

```diff
@@ -0,0 +1,110 @@
+import os
+import shutil
+try:
+    import configparser
+except ImportError:
+        import ConfigParser as configparser
+
+
+from abc import ABCMeta, abstractmethod
+from linchpin.exceptions import LinchpinError
+
+
+class Fetch(object):
+    __metaclass__ = ABCMeta
+
+    def __init__(self, ctx, fetch_type, dest, root):
+        self.ctx = ctx
+        self.fetch_type = fetch_type
+        self.root = root
+        self.tempdirs = []
+        self.dest = os.path.abspath(os.path.realpath(dest))
+        self.config_path = os.path.abspath(os.path.join(os.path.expanduser('~'),
+                                           '.cache/linchpin/fetch.conf'))
+        self.cfgs = self.read_cfg()
+
+    @abstractmethod
+    def fetch_files(self):
+        pass
+
+    def copy_files(self):
+        if self.fetch_type == 'workspace':
+            for path in self.tempdirs:
+                self.copy_dir(path, self.dest)
+        else:
+            self.transfer_section(self.fetch_type)
+
+
+    def transfer_section(self, section):
+        dest_dir = os.path.join(self.dest, section)
+        dir_exists = True
+        if section not in os.listdir(self.dest):
+            dir_exists = False
+            os.mkdir(dest_dir)
+
+        for path in self.tempdirs:
+            src_dir = os.path.join(path, section)
+            if not os.path.exists(src_dir):
+                if not dir_exists:
+                    shutil.rmtree(dest_dir)
+                raise LinchpinError('The {0} directory does not exist in '
+                                    '{1}'.format(self.fetch_type, self.src))
+            self.copy_dir(src_dir, dest_dir)
+
+    def copy_dir(self, src, dest):
+        for root, dirs, files in os.walk(src):
+            files = [f for f in files if not f[0] == '.']
```

Comment:
Thanks! I can't take much credit on that one though, stack overflow helped me out on that one haha